lib/puppetfile-resolver/spec_searchers/forge.rb

```diff
@@ -52,11 +55,25 @@ def self.fetch_all_module_releases(owner, name, forge_api_url = DEFAULT_FORGE_UR
           http_options[:ca_file] = PuppetfileResolver::Util.static_ca_cert_file if ENV['SSL_CERT_FILE'].nil?
 
           response = nil
-          Net::HTTP.start(uri.host, uri.port, http_options) do |http|
-            request = Net::HTTP::Get.new uri
-            response = http.request request
+
+          ::PuppetfileResolver::Util.with_proxy(forge_proxy) do
+            Net::HTTP.start(uri.host, uri.port, http_options) do |http|
+              request = Net::HTTP::Get.new uri
+              response = http.request request
+            end
+          rescue => e
+            msg = "Unable to find module #{owner}-#{name} on #{forge_api_url}"
+            msg += forge_proxy ? " with proxy #{forge_proxy}: " : ": "
+            msg += e.message
+            raise msg
+          end
```

Comment:
Rather annoyingly, net-http raises an error when it gets a 403 Tunnel Forbidden response, instead of returning a response object...